src/Voron/Data/CompactTrees/PersistentDictionary.cs

```diff
@@ -0,0 +1,242 @@
+        }
+    }
+
+    public unsafe partial class PersistentDictionary 
+    {
+        private readonly Page _page;
+        
+        public const int UsableDictionarySize = NumberOfPagesForDictionary * Constants.Storage.PageSize - PageHeader.SizeOf - PersistentDictionaryHeader.SizeOf;
+
+        public long PageNumber => _page.PageNumber;
+        
+        private readonly HopeEncoder<Encoder3Gram<NativeMemoryEncoderState>> _encoder;
+        private byte[] _tempBuffer;
+
+        public static long CreateDefault(LowLevelTransaction llt)
+        {
+            Slice.From(llt.Allocator, $"{nameof(PersistentDictionary)}.Default", out var defaultKey);
+
+            long pageNumber;
+
+            var result = llt.RootObjects.DirectRead(defaultKey);
+            if (result != null)
+            {
+                pageNumber = *(long*)result;
+            }
+            else
+            {
+                var p = llt.AllocatePage(NumberOfPagesForDictionary);
+                p.Flags = PageFlags.Overflow;
+                p.OverflowSize = NumberOfPagesForDictionary * Constants.Storage.PageSize;
+
+                PersistentDictionaryHeader* header = (PersistentDictionaryHeader*)p.DataPointer;
+                header->TableSize = UsableDictionarySize;
+                header->TableMaxEntries = MaxDictionaryEntries;
+                header->CurrentId = p.PageNumber;
+                header->PreviousId = 0;
+
+                // We retrieve the embeeded file from the assembly, copy and checksum the entire thing.             
+                var embeddedFile = typeof(PersistentDictionary).Assembly.GetManifestResourceStream($"Voron.Data.CompactTrees.dictionary.bin");
+                if (embeddedFile == null)
+                    VoronUnrecoverableErrorException.Raise(llt.Environment.Options, "The default dictionary has not been included in the build, the build process needs to be corrected.");
+
+                var dictionary = new byte[embeddedFile.Length];
+                embeddedFile.Read(dictionary);
+
+                var arrayAsInt = MemoryMarshal.Cast<byte, int>(dictionary.AsSpan());
+                int tableSize = arrayAsInt[0];
+                if (tableSize != DefaultDictionaryTableSize)
+                    VoronUnrecoverableErrorException.Raise(llt.Environment.Options, "There is an inconsistency between the expected size of the default compression dictionary and the one read from storage.");
+
+                dictionary.AsSpan()
+                    .Slice(4) // Discard the table size.
+                    .CopyTo(new Span<byte>(p.DataPointer + sizeof(PersistentDictionaryHeader), UsableDictionarySize));
+
+                header->TableHash = XXHash64.Calculate(p.DataPointer + sizeof(long), UsableDictionarySize + PersistentDictionaryHeader.SizeOf - sizeof(long));
+
+#if DEBUG
+                VerifyTable(p);
```

Comment:
Maybe let's decorate `VerifyTable()` method with `[Conditional("DEBUG")]` attribute? I see we always use it inside `#if DEBUG` anyway

Reply:
I was planning to use when performing recovery operations, that's why I didnt mark it. But, until we do there is no reason why we cant.